### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by KateMoreva**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,204 +1,204 @@
...
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>16.0.2</version>
+        </dependency>
+        <dependency>
+            <groupId>com.formdev</groupId>
+            <artifactId>flatlaf</artifactId>
+            <version>1.0</version>
+        </dependency>
+        <dependency>
```
Comment (constantinpape) [inline pom.xml:193]: Do we need an imagej dependency in this repository? We don't want any of the UI functionality here, so maybe we don't need it? cc @K-Meech @tischi

**Review by constantinpape — COMMENTED** [review]
I haven't looked at the changes in much detail, but overall the refactoring looks very good. I am not quite sure if we need the imageJ dependency, see comment above.

Also, I think the package structure could be improved a bit:
- I don't think `.../cba/n5` is a very good "root" for this package, something like `.../cba/bigdataviewer-io` or `bdvio` would be better, see also #17. (I guess this would require some more changes in the MoBIE repo, but I think it's worth it because I would find this confusing).
- It would be nice to have the same structure for the different formats, i.e. have folders `n5`, `openOrganelle` and `ome/zarr` and then have the same structure in each folder.

cc @K-Meech @tischi

File: pom.xml (hunk 1 above)
Comment (KateMoreva) [reply pom.xml:193]: Thank you, I missed this extra dependency.

Comment (constantinpape) [reply pom.xml:193]: Note that you are still using it here: https://github.com/KateMoreva/bigdataviewer-ome-zarr/blob/refactoring/src/main/java/de/embl/cba/n5/Main.java. But I don't think we need that main class.

Comment (KateMoreva) [reply pom.xml:193]: ` imagej` dependency is needed for current tests.

**Comment by constantinpape** [comment]
@KateMoreva regarding the refactoring I mentioned above: I think it's best if we don't do this now and go ahead and merge this first and see that we get it working with MoBIE. I will create an issue about the further refactoring ideas so we can keep track of this.

So please go ahead and merge this whenever you think this is ready (note that there are still some merge conflicts to be fixed.)

**Comment by KateMoreva** [comment]
> @KateMoreva regarding the refactoring I mentioned above: I think it's best if we don't do this now and go ahead and merge this first and see that we get it working with MoBIE. I will create an issue about the further refactoring ideas so we can keep track of this.
> 
> So please go ahead and merge this whenever you think this is ready (note that there are still some merge conflicts to be fixed.)

@constantinpape, before merging this variant we need to merge MoBIE with the dependency on this repo (https://github.com/mobie/mobie-viewer-fiji/pull/405).

**Comment by constantinpape** [comment]
> > 
> 
> @constantinpape, before merging this variant we need to merge MoBIE with the dependency on this repo ([mobie/mobie-viewer-fiji#405](https://github.com/mobie/mobie-viewer-fiji/pull/405)).

Ah ok, now I understand.

**Comment by KateMoreva** [comment]
> > > 
> > 
> > 
> > @constantinpape, before merging this variant we need to merge MoBIE with the dependency on this repo ([mobie/mobie-viewer-fiji#405](https://github.com/mobie/mobie-viewer-fiji/pull/405)).
> 
> Ah ok, now I understand.

Sorry for the uncertainty that I made :)

**Comment by tischi** [comment]
@KateMoreva I tried to glance over it but it is just too much. I trust you that it is better than what we had before ❤️ . I would suggest to just merge it and than further improve it on the main branch. => From me side: please merge.

**Comment by KateMoreva** [comment]
> @KateMoreva I tried to glance over it but it is just too much. I trust you that it is better than what we had before  . I would suggest to just merge it and than further improve it on the main branch. => From me side: please merge.

Ok, thank you! Sorry for the lots of files, should have split them into separate PRs.

**Pull request merged**